src/lib/containers/entity_finder/tree/TreeNode.tsx

```diff
@@ -168,6 +180,7 @@ export const TreeNode: React.FunctionComponent<TreeNodeProps> = ({
           />
         )}
       </div>
+      {/* We hide the children using CSS rather than un/mounting the node on toggle because we want to preserve the child's state if shown, hidden, and shown again */}
```

Comment:
No behavior change, just leaving a note in case someone thinks they can optimize this and tries to change it (which I totally did even though I wrote this code in the first place...)